test/syscalls/linux/pty.cc

```diff
@@ -1640,6 +1640,57 @@ TEST_F(JobControlTest, DISABLED_SetForegroundProcessGroup) {
   ASSERT_NO_ERRNO(res);
 }
 
+// This test verify if a SIGTTOU signal is sent to the calling process'group
+// when tcsetpgrp is called by a background process
+TEST_F(JobControlTest, SetForegroundProcessGroupSIGTTOUBackground) {
+  auto res = RunInChild([=]() {
+    setsid();
```

Comment:
All the calls to `setsid()`, `setpgid()`, and `tcsetpgrp()` should be checked via `TEST_PCHECK()`. Both here and in the below test.